pkg/operator/controller/certificate/default_cert.go

```diff
@@ -17,67 +17,77 @@ import (
 
 // ensureDefaultCertificateForIngress creates or deletes an operator-generated
 // default certificate for a given IngressController as appropriate.  Returns true
-// if it creates, updates, or deletes the secret for the certificate, false
-// otherwise.
+// if it the secret exists, or false if it does not, as well as any errors.
 func (r *reconciler) ensureDefaultCertificateForIngress(caSecret *corev1.Secret, namespace string, deploymentRef metav1.OwnerReference, ci *operatorv1.IngressController) (bool, error) {
 	ca, err := crypto.GetCAFromBytes(caSecret.Data["tls.crt"], caSecret.Data["tls.key"])
 	if err != nil {
 		return false, fmt.Errorf("failed to get CA from secret %s/%s: %v", caSecret.Namespace, caSecret.Name, err)
 	}
-	desired, err := desiredRouterDefaultCertificateSecret(ca, namespace, deploymentRef, ci)
+	wantCert, desired, err := desiredRouterDefaultCertificateSecret(ca, namespace, deploymentRef, ci)
 	if err != nil {
 		return false, err
 	}
-	current, err := r.currentRouterDefaultCertificate(ci, namespace)
+	if !wantCert {
+		// If the operator generated certificate is not being used, ensure that the ingress controller's
+		// Spec.DefaultCertificate secret exists before deleting the operator generated secret.
+		// See https://bugzilla.redhat.com/show_bug.cgi?id=1887441
+		err := r.lookupUserSpecifiedRouterDefaultCertificate(ci, namespace)
```

Comment:
It might be better to move the call to `lookupUserSpecifiedRouterDefaultCertificate` to `desiredRouteDefaultCertificateSecret` https://github.com/openshift/cluster-ingress-operator/pull/505/files#diff-16d9c01d5eac10bd8a3012d74ef5feb542287746e676beca44479eb80cbaa124R79

Reply:
Leaving the call to `r.lookupUserSpecifiedRouterDefaultCertificate` here would save us some API calls, so I think I would prefer to not refactor `desiredRouterDefaultCertificateSecret` in this case.

Reply:
I think I would prefer not to conflate `desiredRouterDefaultCertificateSecret` with logic that checks if the user specified secret exists.

Reply:
I may have misunderstood the comments in https://github.com/openshift/cluster-ingress-operator/pull/505/files#diff-16d9c01d5eac10bd8a3012d74ef5feb542287746e676beca44479eb80cbaa124R144-R146

I was supposing that if you're checking `ci.Spec.DefaultCertificate != nil` in `desiredRouteDefaultCertificateSecret` then you would also want to perform `lookupUserSpecifiedRouterDefaultCertificate` there too (or instead).